File: 2190-count-common-words-with-one-occurrence/count-common-words-with-one-occurrence.java
class Solution {
    public int countWords(String[] words1, String[] words2) {
        Map<String, Integer> map1 = new HashMap<>();
        Map<String, Integer> map2 = new HashMap<>();
        int n1 = words1.length;
        int n2 = words2.length;
        //System.out.println("n1="+n1 +",n2="+n2);
        int count = 0;

        for (int i = 0; i < n1; i++) {
            map1.put(words1[i], (map1.getOrDefault(words1[i], 0)) + 1);
        }
        for (int i = 0; i < n2; i++) {
            if(map1.containsKey(words2[i]) && map1.get(words2[i])<=1 )
            {
                map1.put(words2[i],(map1.getOrDefault(words2[i], 0)) - 1);
            }
        }


       // System.out.println("map1="+map1.keySet().size() +",val2="+map2.keySet().size());

         for (String key : map1.keySet()){
            if(map1.get(key)==0)
            {count++;}
         }
         return count;
    }
}

File: 2190-count-common-words-with-one-occurrence/count-common-words-with-one-occurrence.go

func countWords(word1 []string, word2 []string) int {
	var n int =0
var map1 = map[string]int{}
for  s := range word1{
	str := word1[s]
	if val,ok:= map1[str];
	ok{
		map1[str]= val+1
	}else{
		map1[str]= 1
	}
}

for  s := range word2{
	str := word2[s]
	if value,ok :=map1[str]
	ok{
		if(value<=1){
			map1[str]=value-1
		}
	}
}
for _,value:= range map1{
	if(value==0){
		n=n+1
	}
}

	return  n

}
